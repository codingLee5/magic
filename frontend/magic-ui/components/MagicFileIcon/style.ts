import { createStyles } from "antd-style"

export const useStyles = createStyles(() => ({
	fileIcon: {
		display: "inline-flex",
		alignItems: "center",
		justifyContent: "center",
	},
	image: {
		width: "100%",
		height: "100%",
		objectFit: "contain",
		objectPosition: "center",
	},
}))
