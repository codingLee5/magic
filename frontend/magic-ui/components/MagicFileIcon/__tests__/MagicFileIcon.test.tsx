import { render } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"
import MagicFileIcon from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicFileIcon", () => {
	it("应该正常渲染", () => {
		renderWithTheme(<MagicFileIcon type="pdf" />)
		expect(true).toBe(true)
	})

	// 快照测试
	describe("快照测试", () => {
		it("PDF文件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicFileIcon type="pdf" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("Word文件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicFileIcon type="word" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("Excel文件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicFileIcon type="excel" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("图片文件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicFileIcon type="image" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("视频文件图标快照", () => {
			const { asFragment } = renderWithTheme(<MagicFileIcon type="video" />)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带自定义样式文件图标快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicFileIcon type="pdf" style={{ width: "32px", height: "32px" }} />,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带类名文件图标快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicFileIcon type="pdf" className="custom-file-icon" />,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
