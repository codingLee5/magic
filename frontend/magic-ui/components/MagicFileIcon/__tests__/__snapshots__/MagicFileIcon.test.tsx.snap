// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicFileIcon > 快照测试 > Excel文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="excel"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%2379878F'/%3e%3cpath%20id='Vector_2'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3cpath%20id='Vector_3'%20d='M13.12%208.603C14.657%208.603%2015.888%209.894%2015.888%2011.469C15.888%2012.986%2014.747%2014.239%2013.29%2014.329L13.12%2014.335H12.375C12.0655%2014.3349%2011.767%2014.4495%2011.5372%2014.6568C11.3074%2014.864%2011.1627%2015.1492%2011.131%2015.457L11.125%2015.585V17.065C11.1247%2017.2549%2011.0523%2017.4377%2010.9225%2017.5763C10.7927%2017.715%2010.6151%2017.7992%2010.4256%2017.8121C10.2361%2017.8249%2010.0488%2017.7653%209.90151%2017.6454C9.75421%2017.5255%209.65789%2017.3542%209.632%2017.166L9.625%2017.064V15.585C9.62497%2014.8847%209.89209%2014.2108%2010.3719%2013.7007C10.8517%2013.1906%2011.508%2012.8828%2012.207%2012.84L12.375%2012.835H13.121C13.812%2012.835%2014.388%2012.231%2014.388%2011.469C14.388%2010.755%2013.882%2010.179%2013.249%2010.11L13.121%2010.103H10.375C10.185%2010.1029%2010.0021%2010.0308%209.86321%209.90103C9.72436%209.7713%209.63993%209.5937%209.62697%209.40412C9.61402%209.21454%209.67352%209.02711%209.79343%208.8797C9.91335%208.73229%2010.0847%208.6359%2010.273%208.61L10.375%208.603H13.121H13.12ZM10.376%2018.65C10.5572%2018.65%2010.7323%2018.7156%2010.8689%2018.8348C11.0055%2018.9539%2011.0944%2019.1184%2011.119%2019.298L11.126%2019.4V19.441C11.1259%2019.631%2011.0538%2019.8139%2010.924%2019.9528C10.7943%2020.0916%2010.6167%2020.1761%2010.4271%2020.189C10.2375%2020.202%2010.0501%2020.1425%209.9027%2020.0226C9.75529%2019.9027%209.6589%2019.7312%209.633%2019.543L9.626%2019.441V19.401C9.626%2019.2021%209.70502%2019.0113%209.84567%2018.8707C9.98632%2018.73%2010.1771%2018.65%2010.376%2018.65Z'%20fill='white'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > PDF文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="pdf"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%23FF414C'/%3e%3cpath%20id='Vector_2'%20d='M11.948%209.184C12.521%209.207%2012.976%209.583%2013.091%2010.225C13.152%2010.523%2013.171%2010.845%2013.154%2011.193C13.13%2011.654%2013.05%2012.112%2012.901%2012.739C12.8868%2012.799%2012.8725%2012.859%2012.858%2012.919L12.821%2013.069C13.223%2014.116%2013.789%2014.829%2014.629%2015.346C15.1981%2015.2674%2015.7725%2015.2333%2016.347%2015.244C17.085%2015.268%2017.63%2015.407%2017.926%2015.698C18.212%2015.983%2018.318%2016.5%2018.184%2016.946C18.032%2017.453%2017.612%2017.776%2016.991%2017.776C16.237%2017.776%2015.119%2017.436%2014.151%2016.918C13.594%2017.012%2012.981%2017.178%2012.172%2017.442C11.986%2017.502%2011.797%2017.566%2011.537%2017.654L11.125%2017.793L10.986%2017.84L10.744%2017.92C9.60999%2019.866%208.58399%2020.727%207.74899%2020.662C7.58966%2020.6489%207.4355%2020.5992%207.29849%2020.5168C7.16148%2020.4344%207.0453%2020.3216%206.95899%2020.187C6.79699%2019.947%206.72599%2019.653%206.77299%2019.429C6.86999%2018.944%207.22199%2018.483%207.78599%2018.035C8.21299%2017.695%208.72199%2017.389%209.37599%2017.052C9.86499%2016.192%2010.594%2014.649%2011.146%2013.332C10.713%2011.654%2010.593%2010.67%2010.797%209.922C10.923%209.46%2011.414%209.162%2011.948%209.184ZM9.71799%2017.504L9.33199%2017.864L9.01399%2018.148C8.84899%2018.298%208.70799%2018.431%208.57199%2018.568C8.34715%2018.7876%208.14481%2019.0292%207.96799%2019.289C7.88299%2019.419%207.81799%2019.543%207.77399%2019.661L7.76999%2019.673V19.686C7.76999%2019.714%207.78699%2019.74%207.81399%2019.752C7.82602%2019.7569%207.8391%2019.7586%207.85199%2019.757C7.86574%2019.7556%207.87921%2019.7523%207.89199%2019.747C7.92344%2019.7331%207.95356%2019.7164%207.98199%2019.697C8.06799%2019.642%208.17499%2019.555%208.29599%2019.437C8.62599%2019.119%209.01199%2018.63%209.43799%2017.951L9.71799%2017.504ZM11.978%2014.018L11.924%2014.158C11.499%2015.281%2011.104%2016.174%2010.677%2016.918L10.581%2017.086L10.762%2017.016C10.97%2016.936%2011.793%2016.6%2011.718%2016.63C12.438%2016.34%2012.964%2016.164%2013.528%2016.048L13.68%2016.016L13.556%2015.924C12.979%2015.497%2012.454%2014.884%2012.05%2014.15L11.978%2014.018ZM17.22%2016.342C16.888%2016.22%2016.317%2016.175%2015.51%2016.235L15.227%2016.256L15.489%2016.368C16.519%2016.808%2017.104%2016.903%2017.376%2016.796C17.4035%2016.7862%2017.429%2016.7713%2017.451%2016.752C17.4857%2016.7289%2017.51%2016.6932%2017.5187%2016.6525C17.5275%2016.6118%2017.5201%2016.5693%2017.498%2016.534C17.456%2016.459%2017.362%2016.394%2017.22%2016.342ZM11.974%209.968C11.808%209.968%2011.766%2010.284%2011.791%2010.809C11.813%2011.271%2011.891%2011.783%2011.999%2012.123L12.047%2012.275L12.125%2012.136C12.252%2011.911%2012.33%2011.397%2012.322%2010.904C12.314%2010.342%2012.207%209.968%2011.974%209.968Z'%20fill='white'/%3e%3cpath%20id='Vector_3'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > Word文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="word"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%2379878F'/%3e%3cpath%20id='Vector_2'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3cpath%20id='Vector_3'%20d='M13.12%208.603C14.657%208.603%2015.888%209.894%2015.888%2011.469C15.888%2012.986%2014.747%2014.239%2013.29%2014.329L13.12%2014.335H12.375C12.0655%2014.3349%2011.767%2014.4495%2011.5372%2014.6568C11.3074%2014.864%2011.1627%2015.1492%2011.131%2015.457L11.125%2015.585V17.065C11.1247%2017.2549%2011.0523%2017.4377%2010.9225%2017.5763C10.7927%2017.715%2010.6151%2017.7992%2010.4256%2017.8121C10.2361%2017.8249%2010.0488%2017.7653%209.90151%2017.6454C9.75421%2017.5255%209.65789%2017.3542%209.632%2017.166L9.625%2017.064V15.585C9.62497%2014.8847%209.89209%2014.2108%2010.3719%2013.7007C10.8517%2013.1906%2011.508%2012.8828%2012.207%2012.84L12.375%2012.835H13.121C13.812%2012.835%2014.388%2012.231%2014.388%2011.469C14.388%2010.755%2013.882%2010.179%2013.249%2010.11L13.121%2010.103H10.375C10.185%2010.1029%2010.0021%2010.0308%209.86321%209.90103C9.72436%209.7713%209.63993%209.5937%209.62697%209.40412C9.61402%209.21454%209.67352%209.02711%209.79343%208.8797C9.91335%208.73229%2010.0847%208.6359%2010.273%208.61L10.375%208.603H13.121H13.12ZM10.376%2018.65C10.5572%2018.65%2010.7323%2018.7156%2010.8689%2018.8348C11.0055%2018.9539%2011.0944%2019.1184%2011.119%2019.298L11.126%2019.4V19.441C11.1259%2019.631%2011.0538%2019.8139%2010.924%2019.9528C10.7943%2020.0916%2010.6167%2020.1761%2010.4271%2020.189C10.2375%2020.202%2010.0501%2020.1425%209.9027%2020.0226C9.75529%2019.9027%209.6589%2019.7312%209.633%2019.543L9.626%2019.441V19.401C9.626%2019.2021%209.70502%2019.0113%209.84567%2018.8707C9.98632%2018.73%2010.1771%2018.65%2010.376%2018.65Z'%20fill='white'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > 图片文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="image"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%2379878F'/%3e%3cpath%20id='Vector_2'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3cpath%20id='Vector_3'%20d='M13.12%208.603C14.657%208.603%2015.888%209.894%2015.888%2011.469C15.888%2012.986%2014.747%2014.239%2013.29%2014.329L13.12%2014.335H12.375C12.0655%2014.3349%2011.767%2014.4495%2011.5372%2014.6568C11.3074%2014.864%2011.1627%2015.1492%2011.131%2015.457L11.125%2015.585V17.065C11.1247%2017.2549%2011.0523%2017.4377%2010.9225%2017.5763C10.7927%2017.715%2010.6151%2017.7992%2010.4256%2017.8121C10.2361%2017.8249%2010.0488%2017.7653%209.90151%2017.6454C9.75421%2017.5255%209.65789%2017.3542%209.632%2017.166L9.625%2017.064V15.585C9.62497%2014.8847%209.89209%2014.2108%2010.3719%2013.7007C10.8517%2013.1906%2011.508%2012.8828%2012.207%2012.84L12.375%2012.835H13.121C13.812%2012.835%2014.388%2012.231%2014.388%2011.469C14.388%2010.755%2013.882%2010.179%2013.249%2010.11L13.121%2010.103H10.375C10.185%2010.1029%2010.0021%2010.0308%209.86321%209.90103C9.72436%209.7713%209.63993%209.5937%209.62697%209.40412C9.61402%209.21454%209.67352%209.02711%209.79343%208.8797C9.91335%208.73229%2010.0847%208.6359%2010.273%208.61L10.375%208.603H13.121H13.12ZM10.376%2018.65C10.5572%2018.65%2010.7323%2018.7156%2010.8689%2018.8348C11.0055%2018.9539%2011.0944%2019.1184%2011.119%2019.298L11.126%2019.4V19.441C11.1259%2019.631%2011.0538%2019.8139%2010.924%2019.9528C10.7943%2020.0916%2010.6167%2020.1761%2010.4271%2020.189C10.2375%2020.202%2010.0501%2020.1425%209.9027%2020.0226C9.75529%2019.9027%209.6589%2019.7312%209.633%2019.543L9.626%2019.441V19.401C9.626%2019.2021%209.70502%2019.0113%209.84567%2018.8707C9.98632%2018.73%2010.1771%2018.65%2010.376%2018.65Z'%20fill='white'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > 带类名文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666 custom-file-icon"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="pdf"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%23FF414C'/%3e%3cpath%20id='Vector_2'%20d='M11.948%209.184C12.521%209.207%2012.976%209.583%2013.091%2010.225C13.152%2010.523%2013.171%2010.845%2013.154%2011.193C13.13%2011.654%2013.05%2012.112%2012.901%2012.739C12.8868%2012.799%2012.8725%2012.859%2012.858%2012.919L12.821%2013.069C13.223%2014.116%2013.789%2014.829%2014.629%2015.346C15.1981%2015.2674%2015.7725%2015.2333%2016.347%2015.244C17.085%2015.268%2017.63%2015.407%2017.926%2015.698C18.212%2015.983%2018.318%2016.5%2018.184%2016.946C18.032%2017.453%2017.612%2017.776%2016.991%2017.776C16.237%2017.776%2015.119%2017.436%2014.151%2016.918C13.594%2017.012%2012.981%2017.178%2012.172%2017.442C11.986%2017.502%2011.797%2017.566%2011.537%2017.654L11.125%2017.793L10.986%2017.84L10.744%2017.92C9.60999%2019.866%208.58399%2020.727%207.74899%2020.662C7.58966%2020.6489%207.4355%2020.5992%207.29849%2020.5168C7.16148%2020.4344%207.0453%2020.3216%206.95899%2020.187C6.79699%2019.947%206.72599%2019.653%206.77299%2019.429C6.86999%2018.944%207.22199%2018.483%207.78599%2018.035C8.21299%2017.695%208.72199%2017.389%209.37599%2017.052C9.86499%2016.192%2010.594%2014.649%2011.146%2013.332C10.713%2011.654%2010.593%2010.67%2010.797%209.922C10.923%209.46%2011.414%209.162%2011.948%209.184ZM9.71799%2017.504L9.33199%2017.864L9.01399%2018.148C8.84899%2018.298%208.70799%2018.431%208.57199%2018.568C8.34715%2018.7876%208.14481%2019.0292%207.96799%2019.289C7.88299%2019.419%207.81799%2019.543%207.77399%2019.661L7.76999%2019.673V19.686C7.76999%2019.714%207.78699%2019.74%207.81399%2019.752C7.82602%2019.7569%207.8391%2019.7586%207.85199%2019.757C7.86574%2019.7556%207.87921%2019.7523%207.89199%2019.747C7.92344%2019.7331%207.95356%2019.7164%207.98199%2019.697C8.06799%2019.642%208.17499%2019.555%208.29599%2019.437C8.62599%2019.119%209.01199%2018.63%209.43799%2017.951L9.71799%2017.504ZM11.978%2014.018L11.924%2014.158C11.499%2015.281%2011.104%2016.174%2010.677%2016.918L10.581%2017.086L10.762%2017.016C10.97%2016.936%2011.793%2016.6%2011.718%2016.63C12.438%2016.34%2012.964%2016.164%2013.528%2016.048L13.68%2016.016L13.556%2015.924C12.979%2015.497%2012.454%2014.884%2012.05%2014.15L11.978%2014.018ZM17.22%2016.342C16.888%2016.22%2016.317%2016.175%2015.51%2016.235L15.227%2016.256L15.489%2016.368C16.519%2016.808%2017.104%2016.903%2017.376%2016.796C17.4035%2016.7862%2017.429%2016.7713%2017.451%2016.752C17.4857%2016.7289%2017.51%2016.6932%2017.5187%2016.6525C17.5275%2016.6118%2017.5201%2016.5693%2017.498%2016.534C17.456%2016.459%2017.362%2016.394%2017.22%2016.342ZM11.974%209.968C11.808%209.968%2011.766%2010.284%2011.791%2010.809C11.813%2011.271%2011.891%2011.783%2011.999%2012.123L12.047%2012.275L12.125%2012.136C12.252%2011.911%2012.33%2011.397%2012.322%2010.904C12.314%2010.342%2012.207%209.968%2011.974%209.968Z'%20fill='white'/%3e%3cpath%20id='Vector_3'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > 带自定义样式文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 32px; height: 32px;"
  >
    <img
      alt="pdf"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%23FF414C'/%3e%3cpath%20id='Vector_2'%20d='M11.948%209.184C12.521%209.207%2012.976%209.583%2013.091%2010.225C13.152%2010.523%2013.171%2010.845%2013.154%2011.193C13.13%2011.654%2013.05%2012.112%2012.901%2012.739C12.8868%2012.799%2012.8725%2012.859%2012.858%2012.919L12.821%2013.069C13.223%2014.116%2013.789%2014.829%2014.629%2015.346C15.1981%2015.2674%2015.7725%2015.2333%2016.347%2015.244C17.085%2015.268%2017.63%2015.407%2017.926%2015.698C18.212%2015.983%2018.318%2016.5%2018.184%2016.946C18.032%2017.453%2017.612%2017.776%2016.991%2017.776C16.237%2017.776%2015.119%2017.436%2014.151%2016.918C13.594%2017.012%2012.981%2017.178%2012.172%2017.442C11.986%2017.502%2011.797%2017.566%2011.537%2017.654L11.125%2017.793L10.986%2017.84L10.744%2017.92C9.60999%2019.866%208.58399%2020.727%207.74899%2020.662C7.58966%2020.6489%207.4355%2020.5992%207.29849%2020.5168C7.16148%2020.4344%207.0453%2020.3216%206.95899%2020.187C6.79699%2019.947%206.72599%2019.653%206.77299%2019.429C6.86999%2018.944%207.22199%2018.483%207.78599%2018.035C8.21299%2017.695%208.72199%2017.389%209.37599%2017.052C9.86499%2016.192%2010.594%2014.649%2011.146%2013.332C10.713%2011.654%2010.593%2010.67%2010.797%209.922C10.923%209.46%2011.414%209.162%2011.948%209.184ZM9.71799%2017.504L9.33199%2017.864L9.01399%2018.148C8.84899%2018.298%208.70799%2018.431%208.57199%2018.568C8.34715%2018.7876%208.14481%2019.0292%207.96799%2019.289C7.88299%2019.419%207.81799%2019.543%207.77399%2019.661L7.76999%2019.673V19.686C7.76999%2019.714%207.78699%2019.74%207.81399%2019.752C7.82602%2019.7569%207.8391%2019.7586%207.85199%2019.757C7.86574%2019.7556%207.87921%2019.7523%207.89199%2019.747C7.92344%2019.7331%207.95356%2019.7164%207.98199%2019.697C8.06799%2019.642%208.17499%2019.555%208.29599%2019.437C8.62599%2019.119%209.01199%2018.63%209.43799%2017.951L9.71799%2017.504ZM11.978%2014.018L11.924%2014.158C11.499%2015.281%2011.104%2016.174%2010.677%2016.918L10.581%2017.086L10.762%2017.016C10.97%2016.936%2011.793%2016.6%2011.718%2016.63C12.438%2016.34%2012.964%2016.164%2013.528%2016.048L13.68%2016.016L13.556%2015.924C12.979%2015.497%2012.454%2014.884%2012.05%2014.15L11.978%2014.018ZM17.22%2016.342C16.888%2016.22%2016.317%2016.175%2015.51%2016.235L15.227%2016.256L15.489%2016.368C16.519%2016.808%2017.104%2016.903%2017.376%2016.796C17.4035%2016.7862%2017.429%2016.7713%2017.451%2016.752C17.4857%2016.7289%2017.51%2016.6932%2017.5187%2016.6525C17.5275%2016.6118%2017.5201%2016.5693%2017.498%2016.534C17.456%2016.459%2017.362%2016.394%2017.22%2016.342ZM11.974%209.968C11.808%209.968%2011.766%2010.284%2011.791%2010.809C11.813%2011.271%2011.891%2011.783%2011.999%2012.123L12.047%2012.275L12.125%2012.136C12.252%2011.911%2012.33%2011.397%2012.322%2010.904C12.314%2010.342%2012.207%209.968%2011.974%209.968Z'%20fill='white'/%3e%3cpath%20id='Vector_3'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;

exports[`MagicFileIcon > 快照测试 > 视频文件图标快照 1`] = `
<DocumentFragment>
  <div
    class="acss-nrs666"
    style="width: 24px; height: 24px;"
  >
    <img
      alt="video"
      class="acss-tcohuw"
      src="data:image/svg+xml,%3csvg%20width='25'%20height='24'%20viewBox='0%200%2025%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20id='Group'%3e%3cpath%20id='Vector'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M16.263%200L23.5%207.237V22C23.5%2022.5304%2023.2893%2023.0391%2022.9142%2023.4142C22.5391%2023.7893%2022.0304%2024%2021.5%2024H3.5C2.96957%2024%202.46086%2023.7893%202.08579%2023.4142C1.71071%2023.0391%201.5%2022.5304%201.5%2022V2C1.5%201.46957%201.71071%200.960859%202.08579%200.585786C2.46086%200.210714%202.96957%200%203.5%200L16.263%200Z'%20fill='%2379878F'/%3e%3cpath%20id='Vector_2'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M18.263%207.237C17.7326%207.237%2017.2239%207.02629%2016.8488%206.65121C16.4737%206.27614%2016.263%205.76743%2016.263%205.237V0L23.5%207.237H18.263Z'%20fill='white'%20fill-opacity='0.401'/%3e%3cpath%20id='Vector_3'%20d='M13.12%208.603C14.657%208.603%2015.888%209.894%2015.888%2011.469C15.888%2012.986%2014.747%2014.239%2013.29%2014.329L13.12%2014.335H12.375C12.0655%2014.3349%2011.767%2014.4495%2011.5372%2014.6568C11.3074%2014.864%2011.1627%2015.1492%2011.131%2015.457L11.125%2015.585V17.065C11.1247%2017.2549%2011.0523%2017.4377%2010.9225%2017.5763C10.7927%2017.715%2010.6151%2017.7992%2010.4256%2017.8121C10.2361%2017.8249%2010.0488%2017.7653%209.90151%2017.6454C9.75421%2017.5255%209.65789%2017.3542%209.632%2017.166L9.625%2017.064V15.585C9.62497%2014.8847%209.89209%2014.2108%2010.3719%2013.7007C10.8517%2013.1906%2011.508%2012.8828%2012.207%2012.84L12.375%2012.835H13.121C13.812%2012.835%2014.388%2012.231%2014.388%2011.469C14.388%2010.755%2013.882%2010.179%2013.249%2010.11L13.121%2010.103H10.375C10.185%2010.1029%2010.0021%2010.0308%209.86321%209.90103C9.72436%209.7713%209.63993%209.5937%209.62697%209.40412C9.61402%209.21454%209.67352%209.02711%209.79343%208.8797C9.91335%208.73229%2010.0847%208.6359%2010.273%208.61L10.375%208.603H13.121H13.12ZM10.376%2018.65C10.5572%2018.65%2010.7323%2018.7156%2010.8689%2018.8348C11.0055%2018.9539%2011.0944%2019.1184%2011.119%2019.298L11.126%2019.4V19.441C11.1259%2019.631%2011.0538%2019.8139%2010.924%2019.9528C10.7943%2020.0916%2010.6167%2020.1761%2010.4271%2020.189C10.2375%2020.202%2010.0501%2020.1425%209.9027%2020.0226C9.75529%2019.9027%209.6589%2019.7312%209.633%2019.543L9.626%2019.441V19.401C9.626%2019.2021%209.70502%2019.0113%209.84567%2018.8707C9.98632%2018.73%2010.1771%2018.65%2010.376%2018.65Z'%20fill='white'/%3e%3c/g%3e%3c/svg%3e"
    />
  </div>
</DocumentFragment>
`;
